### DISCUSSION TITLE
Unable to successfully build from a pr with a forked branch.

### DISCUSSION THREAD

**Discussion by sdaconceicao** [jenkinsci/bitbucket-push-and-pull-request-plugin#233] [Q&A]
Hi,

I'm trying to setup this plugin to work in an environment where all prs come from forked branches.  Out of the box I'm only able to get this to work if the branch is part of the main git repo where webhooks have been setup. If I open a pr now with a forked branch, the webhook correctly is received, but the actual build fails as it says the checkout has failed, since it appears to be trying to checkout the changes from the main repo which doesn't have the new commmits.

Is there a configuration step I'm missing for this to work, possibly with the Git Jenkins plugin? I tried a few things, like updating the Branch to build to ${BITBUCKET_SOURCE_BRANCH} but that hasn't helped.

Steps to reproduce

1. Created a git repository in Bitbucket Server.
2. Added webhooks for PR Create and comment added.
3. Created a jenkins pipeline job 
- Push and pull request plugin, and added Pr Create and Comment Added as build triggers. 
- Also added Poll SCM build trigger, and left it blank.
- For Pipeline, using Git SCM, branches to build is left blank.

4. Fork the original git repository, and made it public.
5. Created a branch with changes
6. Open a pr from the fork repo to the original repo

Expected Behavior
PR Builder triggers, checks out the forked code, and runs

Current Behavior
PR Builder triggers, but fails to checkout the forked code.

using GIT_SSH to set credentials 
 > /usr/bin/git fetch --tags --force --progress -- ssh://git@my/reposotoriy.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > /usr/bin/git rev-parse 39879661a001cc82e5a01d07a67fb12d0f8c2a5d^{commit} # timeout=10
ERROR: Checkout failed
hudson.plugins.git.GitException: Command "/usr/bin/git rev-parse 39879661a001cc82e5a01d07a67fb12d0f8c2a5d^{commit}" returned status code 128:
stdout: 39879661a001cc82e5a01d07a67fb12d0f8c2a5d^{commit}

stderr: fatal: ambiguous argument '39879661a001cc82e5a01d07a67fb12d0f8c2a5d^{commit}': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

The same steps above will work if the pr originates from the original repo. It's only when it originates from a fork that it fails.

**Comment by actionjack** [comment]
👍

**Comment by tbugfinder** [comment]
Actually I’m also looking for some feature or workaround which supports forked repos.

**Comment by airdata** [comment]
Same issue here. Any updates how to solve this?
